src/Uno.UI/UI/Xaml/Media/PolyBezierSegment.cs

```diff
@@ -21,11 +21,29 @@ public PointCollection Points
 				typeof(PointCollection), 
 				typeof(PolyBezierSegment), 
 				new FrameworkPropertyMetadata(
-					defaultValue: null, 
+					defaultValue: null,
+					propertyChangedCallback: OnPointsChanged,
 					options: FrameworkPropertyMetadataOptions.AffectsMeasure
 				)
 			);
 
+		private static void OnPointsChanged(DependencyObject dependencyobject, DependencyPropertyChangedEventArgs args)
+		{
+			if (dependencyobject is PolyBezierSegment segment)
+			{
+				if (args.OldValue is PointCollection oldCollection)
+				{
+					oldCollection.UnRegisterChangedListener(segment.OnPointsChanged);
```

Comment:
I would have named the "sub"`OnPointsChanged` differently ... really confusing with the current method.

Reply:
Agreed, I'll change it